File: backend/market_platform/schemas/markets.py
from datetime import date
from decimal import Decimal

from pydantic import BaseModel, Field


class MarketQuoteCard(BaseModel):
    id: str
    symbol: str
    name: str
    kind: str  # index | commodity | fx | sector
    exchange: str | None = None
    value: Decimal | None = None
    change: Decimal | None = None
    change_pct: float | None = None
    sparkline: list[float] = Field(default_factory=list)
    as_of: date | None = None
    # Sector-only extras
    score: Decimal | None = None
    score_change_1w: Decimal | None = None
    return_1m: Decimal | None = None
    rotation_state: str | None = None
    # Canonical sector for ETF → sector stocks click-through
    sector_name: str | None = None


class MarketsOverviewResponse(BaseModel):
    indices: list[MarketQuoteCard] = Field(default_factory=list)
    sector_etfs: list[MarketQuoteCard] = Field(default_factory=list)
    commodities: list[MarketQuoteCard] = Field(default_factory=list)
    sectors: list[MarketQuoteCard] = Field(default_factory=list)
    as_of: date | None = None


class MarketIndicesResponse(BaseModel):
    items: list[MarketQuoteCard] = Field(default_factory=list)
    as_of: date | None = None

File: web/src/types/markets.ts
export type MarketQuoteCard = {
  id: string;
  symbol: string;
  name: string;
  kind: "index" | "commodity" | "fx" | "sector" | string;
  exchange: string | null;
  value: string | number | null;
  change: string | number | null;
  change_pct: number | null;
  sparkline: number[];
  as_of: string | null;
  score: string | number | null;
  score_change_1w: string | number | null;
  return_1m: string | number | null;
  rotation_state: string | null;
  /** Canonical sector for ETF → sector stocks click-through */
  sector_name?: string | null;
};

export type MarketsOverviewResponse = {
  indices: MarketQuoteCard[];
  sector_etfs: MarketQuoteCard[];
  commodities: MarketQuoteCard[];
  sectors: MarketQuoteCard[];
  as_of: string | null;
};

export type MarketEtfsResponse = {
  items: MarketQuoteCard[];
  total: number;
  as_of: string | null;
};
